src/server/services/varen/varen-notifications.test.ts

```diff
@@ -0,0 +1,185 @@
+import MockDate from 'mockdate';
+import { describe, expect, it } from 'vitest';
+
+import {
+  VarenFrontend,
+  VarenRegistratieRederType,
+  VarenVergunningExploitatieType,
+} from './config-and-types';
+import { fetchVarenNotifications } from './varen-notifications';
+import {
+  apiErrorResult,
+  apiSuccessResult,
+} from '../../../universal/helpers/api';
+import { AuthProfileAndToken } from '../../auth/auth-types';
+import * as decos from '../decos/decos-service';
+
+const exploitatieBase_ = {
+  vesselName: 'boatName',
+  id: 'Z-25-0000001',
+  identifier: 'Z/25/0000001',
+  key: 'ABCDEF0123456789ABCDEF0123456789',
+  caseType: 'Varen vergunning exploitatie',
+  title: 'Varen vergunning exploitatie',
+  status: 'In behandeling',
+  decision: 'Nog niet bekend',
+  processed: false,
+  dateRequest: '2025-01-01T00:00:00',
+  statusDates: [],
+  termijnDates: [],
+} satisfies Partial<VarenFrontend>;
+const exploitatieBase =
+  exploitatieBase_ as unknown as VarenVergunningExploitatieType;
+
+describe('Notifications', () => {
+  const authProfileAndToken: AuthProfileAndToken = {
+    profile: { authMethod: 'digid', profileType: 'private', id: '', sid: '' },
+    token: 'xxxxxx',
+  };
+
+  MockDate.set('2025-01-20');
+
+  it('should respond with an error response if fetchDecosZaken returns an error', async () => {
+    vi.spyOn(decos, 'fetchDecosZaken').mockResolvedValueOnce(
+      apiErrorResult('Error', null)
+    );
+
+    const response = await fetchVarenNotifications('x0', authProfileAndToken);
+    const errorResponse = {
+      content: null,
+      message: 'Error fetching Varen zaken data',
+      status: 'ERROR',
+    };
+    expect(response).toStrictEqual(errorResponse);
+  });
+
+  it('should only show most recent notification for every zaak', async () => {
+    const zaakRecent = exploitatieBase;
+
+    const zaakOlderThen3Months = {
+      ...exploitatieBase,
+      dateRequest: '2024-09-01T00:00:00',
+    };
+
+    vi.spyOn(decos, 'fetchDecosZaken').mockResolvedValueOnce(
+      apiSuccessResult([zaakRecent, zaakOlderThen3Months])
+    );
+
+    const { content } = await fetchVarenNotifications(
+      'x3',
+      authProfileAndToken
+    );
+
+    expect(content?.notifications).toHaveLength(1);
+    expect(content?.notifications[0].datePublished).toBe('2025-01-01T00:00:00');
```

Comment:
```suggestion
    expect(content?.notifications[0].datePublished).toBe(zaakRecent.dateRequest);
```
Ik denk dat dit het wat explicieter maakt welke zaak je verwacht. Mooie test